metadata-ingestion/src/datahub/ingestion/source/aws/dynamodb.py

```diff
@@ -174,8 +173,7 @@ def populate_attribute_definitions(self, table) -> Dict[str, str]:
                 attribute_definitions[attr] = new_value
         else:
             table_name = table["TableName"]
-            logger.warn(f"Table {table_name} streaming is NOT enabled.")
-            self.report.report_table_streaming_disabled(table_name)
+            raise RuntimeError(f"Table {table_name} streaming is NOT enabled.")
```

Comment:
Let's also add a TODO to make it configurable whether we want to hard fail or soft fail.